app/scripts/modules/core/src/pipeline/config/stages/overrideTimeout/OverrideTimeout.tsx

```diff
@@ -13,26 +15,35 @@ export interface IOverrideTimeoutConfigProps {
 }
 
 interface IStageConfig {
-  defaultTimeoutMs: number;
+  supportsCustomTimeout: boolean;
 }
 
+const toHoursAndMinutes = (ms: number) => {
+  if (!ms) {
+    return { hours: 0, minutes: 0 };
+  } else {
+    const { hours, minutes } = Duration.fromMillis(ms)
+      .shiftTo('hours', 'minutes')
+      .toObject();
+
+    return {
+      hours: Math.floor(hours),
+      minutes: Math.floor(minutes),
+    };
+  }
+};
+
 export const OverrideTimeout = (props: IOverrideTimeoutConfigProps) => {
   const [hours, setHours] = useState(0);
   const [minutes, setMinutes] = useState(0);
   const [overrideTimeout, setOverrideTimeout] = useState(false);
-  const [configurable, setConfigurable] = useState(false);
-  const [defaults, setDefaults] = useState({ hours: 0, minutes: 0 });
 
   useEffect(() => {
     stageChanged();
-  }, [props.stageConfig, props.stageTimeoutMs]);
+  }, [!!get(props.stageConfig, 'supportsCustomTimeout'), props.stageTimeoutMs]);
```

Comment:
Does hook syntax allow for this? And I guess if it won't change at runtime (a specific stage's ability to support a custom timeout), does it need to be present in this array?

Reply:
yeah, i'm kinda skeptical this even needs to be in here and should probably just remove it entirely. i suppose it partly kept it out of some fear that we were watching changes to the stage config in here for a reason i didn't understand. AFAICT the only thing that should trigger this effect is `stageTimeoutMs` changing.